README.md

```diff
@@ -86,27 +80,126 @@ $POSTGRES_PASSWORD  # postgres's user's password
 ```
 
 If the postgres environment variables are not taking into effect:
+
 ```
 $ docker-compose kill
 $ docker-compose rm
 ```
+
 To access the postgres DB from a workstation:
+
 ```
 $ docker-compose run dev psql -h postgres -U < postgres username from dockerfiles/app.env >
- ```
+```
+
 or from inside the container:
+
 ```
 $ psql -h postgres -U $POSTGRES_USER
 ```
 
+### Running tests
+
+```
+$ docker-compose run --service-ports --rm dev bash
+[root@9eea4caf7740 code]# rake spec
+... tests running ...
+Finished in 0.1669 seconds (files took 0.36978 seconds to load)
+60 examples, 0 failures
+[root@9eea4caf7740 code]# rake
+Migrating to latest
+/usr/local/bin/ruby app.rb
+== Sinatra (v1.4.7) has taken the stage on 4567 for development with backup from Thin
+Thin web server (v1.7.0 codename Dunder Mifflin)
+Maximum connections set to 1024
+Listening on localhost:4567, CTRL+C to stop
+```
+
+You may be prompted to `bundle install`.
+
+## Development Without `docker`
+
+If you do not want to install `docker` and `docker-copmose` and instead want to
+run the service natively you can try installing the following packages:
+
+- `sqlite-devel`
+- `postgresql`
+- `postgresql-devel`
+- `ruby-2.3.0` and `gem`
+- `gem install bundle`
+
+Unfortunately we are not able to help with getting `postgresql` up and running
+on your system, but there's plenty of tutorials that *can* help with that.
+
+If you do have problems installing your system, try reading the
+`dockerfiles/app` file to see how the docker environment is created (it's
+basically a bash script). If you're still having problems setting up your
+environment please feel free to make an issue and we'll help out.
+
+### Setting up the Environment
+
+You'll need to add a few environment variables to your computer. I suggest
+adding the following the lines to your `.bashrc`:
+
+```
+export POSTGRES_PASSWORD=changeme   # your postgres password
+export CACHE_FILE=/tmp/iam-cache    # Where you want to save the cache file to
+export POSTGRES_HOST=127.0.0.1      # Whatever your postgres host ip address is
+```
+
+and reload your shell with `source ~/.bashrc`.
+
+### Running tests
```

Comment:
@athai This is *similar* but as @aaroncohen73 points out this is in the section for running IaM without `docker` so it omits the `docker-compose` commands and finished the **Running IaM Without docker** section.

The rationale here is that people won't read the *entire* document, just the part their interested in. This is an effort to localize information to where people are going to look for it.